### ISSUE TITLE
Does not work out of the box. "The system cannot find the file specified"

### ISSUE THREAD

**Issue by StingyJack** [Microsoft/aspnet-docker#65]
Create a new asp.net mvc web application (no netcore)
Add a local filesytem publish profile
Add Docker Support
Publish or Build
Run `docker build -t microsoft/iis`

Get 
```
Sending build context to Docker daemon  3.584kB
Step 1/4 : FROM microsoft/iis
 ---> b7db4e6fea84
Step 2/4 : ARG source
 ---> Running in 971e5af9f914
 ---> b83d50952dd2
Removing intermediate container 971e5af9f914
Step 3/4 : WORKDIR /inetpub/wwwroot
 ---> b8f3fd7618a9
Removing intermediate container 24c1b86fc566
Step 4/4 : COPY ${source:-obj/Docker/publish} .
COPY failed: GetFileAttributesEx \\?\C:\WINDOWS\TEMP\docker-builder845878484\obj\Docker\publish: The system cannot find the file specified.
```

The documentation appears to weave between netfx and netcore and I cant tell which is which or if it matters. I did try to change the COPY path to `./bin/Release/PublishedOutput` but it still resolves to `C:\WINDOWS\TEMP\docker-builder127245847\bin\Release\PublishedOutput`

I cannot figure out what this wants, but following the tutorial (any of the MSFT ones) should net a working example and they do not. I'm just in a research stage with this, but am ready to exclude Docker entirely from potential cloud solutions with how long a simple example is taking between this and the cryptic Symantec issue. 

Whats the COPY command want? Is docker support broken in VS 2017? Are the tutorials wrong?

**Comment by CESARDELATORRE** [comment]
@StingyJack - VS2017 works with Docker containers, so it must be an issue related to the environment.
Several questions that might be obvious, but just in case:
- I supposed you installed "Docker for Windows" first.
- Since you are using a "FROM Microsoft/iis", that means you are using .NET Framework and IIS, so it needs to use Windows Containers. It cannot be Linux containers. Did you switch to Windows Containers in "Docker for Windows" settings? - By default it uses Linux Containers.
- Did you shared your drive in "Docker for Windows" Settings? (that is C: or D: etc.). That is needed for VS so it'll be able to attach to the container and debug, later on..
- Info: When building the project and having Docker enabled in your project, your binaries should be published within the folder ../yourproject/obj/Docker/publish - So, that path is right.
Then, the COPY command in the dockerfile is trying to copy those binaries into the current folder within the Container.
- Is any new project created with Docker support in VS working for you if you don't "Add a local filesytem publish profile" as you mentioned? - Just trying to isolate the issue.
- Finally, what version of VS2017 are you using?

**Comment by StingyJack** [comment]
Yes, docker for windows with windows containers, right clicked the vs project to add docker support, built project, then tried to run the `docker build` command.  I had gone through some of those steps a bunch of times trying to get past Symantec's nonsense before finally just uninstalling SEP. 

I have been able to use the microsoft/nanoserver image and run a simple copy of a powershell script from one of the first tutorials I found, but VS 15.4 doesnt seem to want to create something I can use.  

Sharing a drive is a new instruction. I'll see if I missed that.

The microsoft/iis image is what the vs tools picked as part of adding docker support. I took that dockerfile as it was at first, and only started twiddling with it when it wasn't working.  It will have to work from a TFS vnext build server using an MSBuild task eventually and not just from VS (cant use VS build task, MSFT seems to have dropped support for TFS 2015 in this area). 

All the projects I would be looking to use have the local file system publish profile already, i suppose it could be removed, but then how does one get the software that originates in cs projects on to an image?

**Comment by CESARDELATORRE** [comment]
I just want to isolate the issue, that's why I'm suggesting not to add the local file system publish and make a regular "hello world" Windows Container work, which should work.

About sharing drives, you can see that is a requirement for VS in the basic getting started page, almost right at the beginning:
**Visual Studio Tools for Docker**
https://docs.microsoft.com/en-us/aspnet/core/publishing/visual-studio-tools-for-docker 

Yes, the image depends on the type of project you are using. If it is ASP.NET Core it would have been a ASPNETCore image based on .NET Core which can run on Linux and Windows (multi-arch).
If it is a traditional .NET Framework project, it'll use the microsoft/iis image or the ASP.NET image, depending on the project.

Here you can see two traditional ASP.NET web apps (Web Forms and MVC) using Windows Containers, too: 
https://github.com/dotnet-architecture/eShopModernizing 
And how to set it up at the Getting Started Wiki:
https://github.com/dotnet-architecture/eShopModernizing/wiki/02.-How-to-containerize-the-.NET-Framework-web-apps-with-Windows-Containers-and-Docker 

Btw, what type of VS project template are you trying to create?

It would be good to isolate the issue. When we do, if it is not a problem related to your environment, I'll send it to the VS Tools for Docker team folks, if it is a real issue.

**Comment by StingyJack** [comment]
I don't think I installed any specific "VS tools for docker" unless it was buried in the workload selections. I just right clicked the project and chose Add, docker support as that's what is step 1 on most of the tutorials.  I will definitely read through that link though. Thanks!

**Comment by Jinhuafei** [comment]
@StingyJack Have you tried on latest VS 15 update? It should be as simple as two steps to run your mvc app inside container in VS. 1. add docker support on your mvc project 2. ctrl+F5

Let us know if you still have any issue.
